### DISCUSSION TITLE
Какой тип таблицы применить для создания отчетов: временную или расчетную в дереве?

### DISCUSSION THREAD

**Discussion by horyachka** [totumonline/totum-ru-issues-and-discussions#33] [Как это сделать?] [answered]
Добрый день!

В моем проекте отправитель и получатель заполняют данные про перемещение тары в **простой** таблице.

#### На основе данных из указанной простой таблицы формируются отчеты:

+ **Отчет: остатки тары:**
![image](https://user-images.githubusercontent.com/91678200/142070815-cae833db-0d26-41c2-a4f0-37f81665dd1c.png)

+ **Отчет: движение тары за период:**
![image](https://user-images.githubusercontent.com/91678200/142072654-547a0510-97ff-46d8-92d3-b9c2f22d74eb.png)

+ **Другие отчеты похожие на отчет "движение тары за период".**
В отчетах нужны граничные условия: дата от + дата до; отправитель, получатель и другие.
Условие: пользователь видит только свои строки в отчетах.

#### Вопрос: какой тип таблицы выбрать для создания отчетов — ВРЕМЕННУЮ или РАСЧЕТНУЮ В ДЕРЕВЕ?

Спасибо!

**Comment by ttm-alexeypolunin** [comment, marked as answer]
### [Видео на YouTube —>](https://youtu.be/7GxMubfswh4)

### [Статья в мастерклассах про архитектуру решений —>](https://docs.totum.online/workshops)

Однозначно **Временную!**

Дело в том, что отчеты будут смотреть разные пользователи одновременно и у каждого пользователя будут свои граничные условия. Если это будет происходит в расчетной, то она будет пересчитываться для всех пользователей.

Особенность временных в том, что они создаются виртуально для каждой вкладки браузера. Таким образом действия одного пользователя во временной никак не влияют на то, что видит в это же время второй пользователь смотрящий в эту же таблицу на своем компе.

[Подробнее про место временнх таблиц в архитектуре решения —>](https://docs.totum.online/workshops)

Во временных нет префильтров, поэтому все граничные условия пишутся в хедер.

Дальше по этим граничным условиям выбираются данные и таблица [автозаполняется](https://docs.totum.online/autofill-tables).

Что бы уменьшить нагрузку, можно поставить `исполнять только при добавлении` коду вычисляющему набор данных для таблицы и сделать кнопку в которой выполнять

```

=: recalculate(table: $#ntn; field: 'h)insert')

```

Параметр field указывает какие коди надо пересчитать даже если они только при добавлении.

![Снимок экрана 2021-11-18 в 16 51 40](https://user-images.githubusercontent.com/55755565/142433637-b77cb830-43dc-498f-b6f4-3fe32f41da0c.png)
